plugins/module_utils/parsing.py

```diff
@@ -4,11 +4,24 @@
 # GNU General Public License v3.0+ (see COPYING.GPLv3 or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import absolute_import, division, print_function
+import re
 
 __metaclass__ = type
 
 METRIC_SUFFIXES = ["K", "M", "G", "T", "P"]
 
+duration_pattern = re.compile(
+    r"^((?P<Y>[1-9]\d*)Y)?((?P<W>[1-9]\d*)W)?((?P<D>[1-9]\d*)D)?(((?P<H>[1-9]\d*)H)?((?P<M>[1-9]\d*)M)?)?$"
```

Comment:
This is impossible to read and hard to update in future. I suggest using some already existing libraries like `datetime` (e.g. function `datetime.strptime`).

Reply:
It doesn't work in our case. `datetime.strptime` requires exact format of date. But we can have different combinations for duration. (e.g 10H5D, 3W5D, etc)